k4FWCore/components/Reader.cpp

```diff
@@ -40,27 +40,24 @@ class CollectionPusher : public Gaudi::Functional::details::BaseClass_t<Gaudi::F
 
   template <typename T>
   using OutputHandle_t = Gaudi::Functional::details::OutputHandle_t<Traits_, std::remove_pointer_t<T>>;
-  std::vector<OutputHandle_t<Out>> m_outputs;
 
 protected:
+  std::vector<OutputHandle_t<Out>> m_outputs;
+
   Gaudi::Property<std::vector<std::string>> m_inputCollections{
-      this, "InputCollections", {"First collection"}, "List of input collections"};
-  // Gaudi::Property<std::string>                                        m_input{this, "Input", "Event", "Input file"};
+      this, "InputCollections", {}, "List of input collections"};
 
 public:
   CollectionPusher(std::string name, ISvcLocator* locator)
```

Comment:
Is there some documentation about what this `CollectionPusher` does?

Reply:
It's the base class of the Reader, it could be a single class and I plan on rewriting it. I think this is how it was done in LHCb by the time it was written, now I'm not sure if there is any purpose, certainly we do not have other readers that could use it.

Reply:
Ok, thanks for the explanation!